src/main/java/org/cru/godtools/api/v2/PhraseResource.java

```diff
@@ -0,0 +1,236 @@
+
+	private static final Logger logger = Logger.getLogger(PhraseResource.class);
+
+	@GET
+	@Produces(MediaType.APPLICATION_JSON)
+	public Response listPhrasesForPage(@HeaderParam("Authorization") String authorization,
+									   @PathParam("package") String packageCode,
+									   @PathParam("pageName") String pageName)
+	{
+		logger.info(String.format("Listing phrases w/ authorization %s", authorization));
+
+		AuthorizationRecord.checkAdminAccess(authorizationService.getAuthorizationRecord(null, authorization), clock.currentDateTime());
+
+		logger.info(String.format("Listing phrases, admin validated.  Listing phrases for %s-%s", packageCode, pageName));
+
+		Optional<GodToolsTranslation> translationOptional = draftTranslation.retrieve(BASE_LANGUAGE_CODE, packageCode, false);
+
+		if(!translationOptional.isPresent())
+		{
+			translationOptional = publishedTranslation.retrieve(BASE_LANGUAGE_CODE, packageCode, false);
+
+			if(!translationOptional.isPresent())
+			{
+				logger.warn("Something weird... no published English translation for: " + packageCode);
+
+				return Response.status(Response.Status.BAD_REQUEST).build();
+			}
+		}
+
+		Optional<PageStructure> pageOptional = translationOptional.get().getPage(pageName);
+
+		if(!pageOptional.isPresent())
+		{
+			logger.warn(String.format("Listing phrases, Page %s not found English translation for %s", pageName, packageCode));
+
+			return Response.status(Response.Status.NOT_FOUND).build();
+		}
+
+		List<TranslationElement> translationElements = translationElementService.selectByTranslationIdPageStructureId(translationOptional.get().getTranslation().getId(),
+				pageOptional.get().getId());
+
+		return Response
+				.ok(translationElements)
+				.build();
+	}
+
+	@POST
+	@Consumes(MediaType.APPLICATION_JSON)
+	public Response addPhraseToPackage(@HeaderParam("Authorization") String authorization,
+									   @PathParam("package") String packageCode,
+									   @PathParam("pageName") String pageName,
+									   TranslationElement translationElement)
+	{
+		logger.info(String.format("Adding phrase w/ authorization %s", authorization));
+
+		AuthorizationRecord.checkAdminAccess(authorizationService.getAuthorizationRecord(null, authorization), clock.currentDateTime());
+
+		logger.info(String.format("Adding phrase, admin validated.  Adding phrase for %s-%s", packageCode, pageName));
+
+		Optional<GodToolsTranslation> translationOptional = draftTranslation.retrieve(BASE_LANGUAGE_CODE, packageCode, false);
+
+		if(!translationOptional.isPresent())
+		{
+			translationOptional = publishedTranslation.retrieve(BASE_LANGUAGE_CODE, packageCode, false);
+
+			if(!translationOptional.isPresent())
+			{
+				logger.warn("Something weird... no published English translation for: " + packageCode);
+
+				return Response.status(Response.Status.BAD_REQUEST).build();
+			}
+		}
+
+		Optional<PageStructure> pageOptional = translationOptional.get().getPage(pageName);
```

Comment:
Good call.. though I had to change 
````java
return Response.status(Response.Status.BAD_REQUEST).build();
````
to
````
throw new NotFoundException();
````
.. otherwise I'd have to different possible return types.  The exception model should work though.

Good suggestion!